File: lektion 3.3/app/src/main/java/com/example/lektion33/MainActivity.java
package com.example.lektion33;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.KeyEvent;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;


public class MainActivity extends AppCompatActivity {

EditText et;
Button btn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

         et = findViewById(R.id.editTextText);
         btn = findViewById(R.id.button);

         et.setOnFocusChangeListener(new View.OnFocusChangeListener() {
             @Override
             public void onFocusChange(View v, boolean hasFocus) {
                 if(!hasFocus)
                 Log.i("Aleksander", "unfocused ");
             }
         });

         et.setOnEditorActionListener(new TextView.OnEditorActionListener() {
             @Override
             public boolean onEditorAction(TextView textView, int i, KeyEvent keyEvent) {

                 return false;
             }
         });

         btn.setOnClickListener((e)->{
             Intent i = new Intent(MainActivity.this, MainActivity2.class);
             i.putExtra("Key", et.getText().toString());
             startActivity(i);
         });



        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
}

File: lektion 6.1/app/src/main/java/com/example/lektion61/MainActivity.kt
package com.example.lektion61

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import org.json.JSONObject

class MainActivity : AppCompatActivity() {

    lateinit var tv:TextView
    lateinit var btn:Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)






        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        tv = findViewById(R.id.textView)
        btn = findViewById(R.id.button)

        btn.setOnClickListener() {
            Log.i("Aleksander", "onCreate: ")


            var rq: RequestQueue = Volley.newRequestQueue(this)

            var url = "https://italian-jokes.vercel.app/api/jokes"

            var request = StringRequest(Request.Method.GET, url, { res ->

                Log.i("Aleksander", "success!!: " + res)
                tv.text = JSONObject(res).getString("joke")

            }, {
                Log.i("Aleksander", "fail!!: ")

            })

            rq.cache.clear()
            rq.add(request)


        }
    }
}
